configure.ac

```diff
@@ -183,32 +195,12 @@ CFLAGS="$CFLAGS_RESTORE"
 AS_IF([test "x$ncurses_cv_wget_wch" != xyes],
     [AC_MSG_ERROR([ncurses does not support wide characters])])
 
-### Check for glib libraries
-PKG_CHECK_MODULES([glib], [glib-2.0 >= 2.62.0], [],
-    [AC_MSG_ERROR([glib 2.62.0 or higher is required for profanity])])
-PKG_CHECK_MODULES([gio], [gio-2.0], [],
-    [AC_MSG_ERROR([libgio-2.0 from glib-2.0 is required for profanity])])
-
-### Check for other profanity dependencies
-AC_SEARCH_LIBS([fmod], [m], [],
-   [AC_MSG_ERROR([math.h is required])], [])
-
-PKG_CHECK_MODULES([curl], [libcurl >= 7.62.0], [],
-    [AC_CHECK_LIB([curl], [main], [],
-        [AC_MSG_ERROR([libcurl is required for profanity])])])
-
-PKG_CHECK_MODULES([SQLITE], [sqlite3 >= 3.22.0], [],
-    [AC_MSG_ERROR([sqlite3 3.22.0 or higher is required for profanity])])
-
-AS_IF([test "x$enable_icons_and_clipboard" != xno],
-    [PKG_CHECK_MODULES([GTK], [gtk+-3.0 >= 3.24.0],
-        [AC_DEFINE([HAVE_GTK], [1], [libgtk module])],
-        [AS_IF([test "x$enable_icons_and_clipboard" = xyes],
-               [PKG_CHECK_MODULES([GTK], [gtk+-2.0 >= 2.24.10],
-                                  [AC_DEFINE([HAVE_GTK], [1], [libgtk module])],
-                                  [AC_MSG_ERROR([gtk+-2.0 >= 2.24.10 or gtk+-3.0 >= 3.24.0 is required for icons and clipboard])],
-                                  [AC_MSG_NOTICE([gtk+-3.0/gtk+2.0 not found, icons and clipboard not enabled])])])])])
+## Check for ncursesw/ncurses.h first, Arch linux uses ncurses.h for ncursesw
+AC_CHECK_HEADERS([ncursesw/ncurses.h], [], [])
+AC_CHECK_HEADERS([ncurses.h], [], [])
+AC_CHECK_HEADERS([curses.h], [], [])
 
+# Check for readline
```

Comment:
Can we replace this with `AX_LIB_READLINE` https://www.gnu.org/software/autoconf-archive/ax_lib_readline.html ? Seems like we are doing more in OSX case.